src/pocketmine/Player.php

```diff
@@ -1840,17 +1863,17 @@ public function handleLogin(LoginPacket $packet) : bool{
 		$this->username = TextFormat::clean($packet->username);
 		$this->displayName = $this->username;
 		$this->iusername = strtolower($this->username);
-
 		if($packet->locale !== null){
 			$this->locale = $packet->locale;
 		}
 
 		if(count($this->server->getOnlinePlayers()) >= $this->server->getMaxPlayers() and $this->kick("disconnectionScreen.serverFull", false)){
 			return true;
 		}
-
 		$this->randomClientId = $packet->clientId;
-
+		$this->deviceId = $packet->clientData['DeviceId'];
+		$this->deviceModel = $packet->clientData['DeviceModel'];
+		$this->deviceOS = $packet->clientData['DeviceOS'];
```

Comment:
Double quotes should be used instead of single quotes

Reply:
`$foo = [];
$bar = [];
for ($i = 0 ; $i <= 1000000 ; $i++) {
	$time1 = ms();
	echo "just a test" , PHP_EOL;
	$time2 = ms();
	$foo[] = $time2 - $time1;
	$time1 = ms();
	echo 'just a test' , PHP_EOL;
	$time2 = ms();
	$bar[] = $time2 - $time1;
}
arsort($foo);
arsort($bar);
echo '----- use double quotes time-----',PHP_EOL , '[minium]' , $foo[count($foo) - 1],PHP_EOL , '[maxium]' , $foo[0],PHP_EOL,'----- use single quotes time-----',PHP_EOL , '[minium]' , $bar[count($foo) - 1],PHP_EOL , '[maxium]' , $bar[0];
`
_----- use double quotes time-----
[minium]2.000000000002E-6
[maxium]3.9999999999984E-5
----- use single quotes time-----
[minium]1.9999999999742E-6
[maxium]2.9999999999752E-6_
i think using single quotes is faster than using double quotes,so why not use it?🤣